building-applications/expression-language.md

```diff
@@ -0,0 +1,73 @@
+| listToListOfStructs(list, field)   | Converts a list of items to a list of maps with one element named after `field` For instance if your list is `'this is a question'` and you call `listToListOfStructs(list, 'text')` the result is a list list `[{"text":"this is a question"}]`. This is very handful when you have to convert a list of texts to a list of structs|
+| uppercase(input)                  | Changes the capitalization of a string. If input is not a string, it attempts a string conversion. If the input is null, it returns null.                                                                                                                                                                                                                                                                                                                                           |
+| toJson                            | Converts `input` to a JSON string.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| trim(input)                       | Returns the input string with all leading and trailing spaces removed. If the input is not a string, it attempts a string conversion. |
+| unpack(input, fieldsList)         | Returns a map containing the elements of `input`, for each field in the `fieldList` you will see an entry in the map. If the input is a string it is converted to a list using the `split()` function with the '`,`' separator |
+| split(input, separatorExpression) | Split the input to a list of strings, this is internally using the String.split() function. An empty input corresponds to an empty list. The input is convered to a String using the str() function. |
+| timestampAdd(input, delta, unit)  | Returns a timestamp formed by adding `delta` in `unit` to the `input` timestamp. `input` is a timestamp to add to. `delta` is a `long` amount of `unit` to add to `input`. Can be a negative value to perform subtraction. `unit` the string unit of time to add or subtract. Can be one of [`years`, `months`, `days`, `hours`, `minutes`, `seconds`, `millis`]. |
+
+
+## Conditional Steps
+
+Each `step` accepts an optional `when` configuration that is evaluated at step execution time against current records (the current step in the transform pipeline).
+
+\
+The `when` condition supports the expression language syntax, which provides access to the record attributes as follows:
```

Comment:
these fields (or part of the record? which is the best word?) are always available now, so this section is not only about the "when" clause.
we can rework this section and say that we an access all of these fields when writing EL expressions